### PR TITLE
[expo-sensors] Pedometer: replace custom listener return type with unimodules Subscription

### PR DESCRIPTION
**Pull request by Simek**
# Why

Refs: #12496

During work on the PR above I have spotted that `Pedometer` uses custom listener return type instead of unimodules `Subscription`. 

I'm not sure if this refactor is correct, but I have decided to push it anyway to make it easier to see the changes and provide the feedback.

# How

This PR removes the `PedometerListener` custom interface and replaces it by unimodules `Subscription` type (just like the `Battery` module code does).

To add the description to the `Subscription` type I have also had to edit and regenerate the `EventEmitter` file (and add description in there). This change also will be required for the `Battery` module conversion.

CC @brentvatne  @bbarthec 

# Test Plan

The changes have been tested on `localhost` using regnerated data for `Pedometer` module which is already using the auto-generated API feature.

Correct type of `remove` function will appear automatically after merge of #12496.

# Preview

Screenshot 2021-04-12 at 20 05 47

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by bbarthec — CHANGES_REQUESTED** [review]
That's a good catch 👏 
I need you to add some CHANGELOG entry for this change as it affects the user-facing API of the module (namely the exported type name).
I believe it might be treated as a `bug fix` (`patch version` bump) or `new feature` (`minot version` bump) (no need to label it as `breaking change` (`major version` bump)

**Comment by Simek** [comment]
@bbarthec Changelog entry added, hope it's bearable, I'm not good at writing such entries. 😉

File: packages/expo-sensors/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Pedometer `watchStepCount` returns an unified Unimodules type now - `Subscription`, instead of the custom interface type. ([#12497](https://github.com/expo/expo/pull/12497) by [@Simek](https://github.com/simek))
```
Comment (bbarthec) [inline packages/expo-sensors/CHANGELOG.md:11]:
```suggestion
- Replaced `Pedometer.watchStepCount()` return type (`PedometerListener`) with an unified Unimodules type - `Subscription`. ([#12497](https://github.com/expo/expo/pull/12497) by [@Simek](https://github.com/simek))
```

**Review by bbarthec — CHANGES_REQUESTED** [review]
I'm not sure about backporting this to `SDK-41`, so please remove backported files 😉

**Review by bbarthec — APPROVED** [review]
LGTM 👍

**Pull request merged**